Parser/tokenizer.c

```diff
@@ -1580,6 +1611,8 @@ tok_get(struct tok_state *tok, const char **p_start, const char **p_end)
                     if (c < '0' || c >= '8') {
                         tok_backup(tok, c);
                         if (isdigit(c)) {
+                            // Move to the actual current token that is incorrect
+                            tok_nextc(tok);
```

Comment:
If you want to call `tok_nextc()` here, there is no need to call `tok_backup()` before it.

Reply:
Is to preserve the syntaxerror in the `else` condition but we can move the backup there